lib/application/widgets/theme/custom_button.dart

```diff
@@ -0,0 +1,313 @@
+          return color.shade800;
+        case ButtonState.disabled:
+          return color.shade800.withOpacity(0.4);
+      }
+    }
+
+    return _CustomElevatedButton(
+      text: text,
+      backgroundColorBuilder: backgroundColorBuilder,
+      backgroundColor: backgroundColor,
+      onPressed: onPressed,
+      leadingAsset: leadingAsset,
+    );
+  }
+}
+
+/// A visually opinionated custom [TextButton].
+///
+/// Visually similar to Material's [TextButton] but with custom disable/pressing behaviors and an optional
+/// [leadingAsset].
+class CustomTextButton extends HookWidget {
+  const CustomTextButton({required this.text, this.color, this.leadingAsset, this.onPressed});
+
+  final String text;
+
+  /// Optional leading asset.
+  ///
+  /// Its color will be overriden by [color] - or the theme primary swatch if `null`.
+  final String? leadingAsset;
+  final VoidCallback? onPressed;
+
+  /// Button's color.
+  ///
+  /// Applies to all button content, including [leadingAsset] if set.
+  final MaterialColor? color;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = useTheme();
+    final buttonColorSwatch = color ?? theme.primarySwatch;
+    final textTheme = Theme.of(context).textTheme.button!;
+
+    Color? buttonColor(ButtonState state) {
+      switch (state) {
+        case ButtonState.normal:
+          return buttonColorSwatch.shade300;
+        case ButtonState.pressed:
+          return buttonColorSwatch.shade400;
+        case ButtonState.disabled:
+          return buttonColorSwatch.shade400.withOpacity(0.4);
+      }
+    }
+
+    Widget leadingAssetBuilder(ButtonState state) => Image.asset(leadingAsset!, color: buttonColor(state));
+
+    return CustomButton(
+      text: text,
+      leadingWidgetBuilder: leadingAsset != null ? leadingAssetBuilder : null,
+      onPressed: onPressed,
+      textStyleBuilder: (state) => textTheme.copyWith(color: buttonColor(state)),
+    );
+  }
+}
+
+/// Wraps a [CustomButton] following a similar style to [ElevatedButton].
+///
+/// Reused by custom buttons implementations.
+/// See also:
+///
+///   * [PrimaryElevatedButton]
+///   * [SecondaryElevatedButton]
+class _CustomElevatedButton extends StatelessWidget {
+  const _CustomElevatedButton({
+    required this.text,
+    required this.backgroundColorBuilder,
+    this.backgroundColor,
+    this.onPressed,
+    this.leadingAsset,
+  });
+
+  final String text;
+
+  /// {@template _CustomElevatedButton.backgroundColor}
+  /// Button's background color swatch.
+  /// {@endtemplate}
+  final MaterialColor? backgroundColor;
+  final VoidCallback? onPressed;
+
+  /// {@template _CustomElevatedButton.leadingAsset}
+  /// Optional leading asset.
+  ///
+  /// Its color will be overriden by the button `TextStyle`.
+  /// {@endtemplate}
+  final String? leadingAsset;
+
+  /// Builds the button background color based on the button state.
+  final Color? Function(ButtonState state) backgroundColorBuilder;
```

Comment:
When will color be null?

Reply:
This is a `_CustomElevatedButton`, how come `CustomTextButton` has anything to do with it?